Tasks/PullRequestCodeAnalysis/prcatask.ts

```diff
@@ -10,38 +12,52 @@ import { TaskLibLogger } from './TaskLibLogger';
-    // Looks like: "Expected message limit to be a number, but instead it was NOT_A_NUMBER"
-    tl.setResult(tl.TaskResult.Failed, tl.loc('Error_NotPullRequest', messageLimitInput));
-}
-
-var tlLogger: TaskLibLogger = new TaskLibLogger();
-var collectionUrl: string = tl.getVariable('System.TeamFoundationCollectionUri');
-var token: string = '';
-var repositoryId: string = tl.getVariable('Build.Repository.Id');
 
-var sourceBranch: string = tl.getVariable('build.sourceBranch');
+var sourceBranch: string = tl.getVariable('Build.SourceBranch');
 
-// Do not continue if the build was not triggered by a pull request
+// Do nothing (except log a message) if the build was not triggered by a pull request
 if (!sourceBranch.startsWith('refs/pull/')) {
     // Looks like: "Skipping pull request commenting - this build was not triggered by a pull request."
     console.log(tl.loc('Error_NotPullRequest'));
 } else {
-    var pullRequestId: number = Number(sourceBranch.replace('refs/pull/', ''));
+
+    var messageLimitInput:string = tl.getInput('messageLimit');
+    var messageLimit: number = Number(messageLimitInput);
     if (isNaN(messageLimit)) {
-        tl.debug(`Expected pull request ID to be a number. Actual: ${sourceBranch.replace('refs/pull/', '')}`);
+        // Looks like: "Expected message limit to be a number, but instead it was NOT_A_NUMBER"
+        tl.setResult(tl.TaskResult.Failed, tl.loc('Error_InvalidMessageLimit', messageLimitInput));
+    }
+
+    var tlLogger: TaskLibLogger = new TaskLibLogger();
+    var collectionUrl: string = tl.getVariable('System.TeamFoundationCollectionUri');
+    var repositoryId: string = tl.getVariable('Build.Repository.Id');
+
+    // Get authentication from the agent itself
+    var auth = tl.getEndpointAuthorization("SYSTEMVSSCONNECTION", false);
+    if (auth.scheme !== "OAuth") {
+        // We cannot get the access token, fail the task
+        // Looks like: "Could not get an authentication token from the build agent."
+        tl.error(tl.loc('Error_FailedToGetAuthToken'));
+        // Looks like: "Pull Request Code Analysis failed."
+        tl.setResult(tl.TaskResult.Failed, tl.loc('Info_ResultFail')); // Set task failure
+    }
+
+    let token = auth.parameters["AccessToken"];
+    var pullRequestId: number = Number.parseInt(sourceBranch.replace('refs/pull/', ''));
+    if (isNaN(pullRequestId)) {
+        tl.debug(`Expected pull request ID to be a number. Attempted to parse: ${sourceBranch.replace('refs/pull/', '')}`);
         // Looks like: "Could not retrieve pull request ID from the server."
         tl.setResult(tl.TaskResult.Failed, tl.loc('Error_InvalidPullRequestId'));
     }
 
     var orchestrator:PrcaOrchestrator = PrcaOrchestrator.CreatePrcaOrchestrator(tlLogger, collectionUrl, token, repositoryId, pullRequestId);
-    orchestrator.postSonarQubeIssuesToPullRequest(process.env)
+    orchestrator.postSonarQubeIssuesToPullRequest(tl.getVariable('PRCA_REPORT_PATH'))
```

Comment:
New task variable "PRCA_REPORT_PATH" should be assigned to by tasks supporting SonarQube analysis. Engineering work is already done in Maven, Gradle remains to do. Future pull requests will address this.